libtransmission/net.h

```diff
@@ -147,22 +148,35 @@ enum tr_address_type : uint8_t
     NUM_TR_AF_INET_TYPES
 };
 
-std::string_view tr_ip_protocol_sv(tr_address_type type) noexcept;
+auto constexpr inline IpProtocolToSv = std::array{ std::string_view{ "IPv4" }, std::string_view{ "IPv6" } };
+static_assert(std::size(IpProtocolToSv) == NUM_TR_AF_INET_TYPES);
```

Comment:
I don't love the idea of making new items in the global namespace without a `tr_` prefix.

If the goal here is clarity & inlining, the old `tr_ip_protocol_sv()` method could be made `constexpr` and either hold this array or have switch statements?

_Maybe_ we should consider namespacing the things in `net.h`? I'm undecided on that approach. It feels like that's where the module is headed long-term but maybe that is overkill?

Reply:
I'll revert to how it was. I was probably just going too far on trying to shorten the code.

Reply:
f58a16a

Reply:
These funcitons likely won't be called in a context where they can actually be evaluated at compile time anyway, so I opted to use the good old declaration and definition.